static/app/views/issueDetails/groupEvents.tsx

```diff
@@ -33,86 +25,59 @@ const excludedTags = [
   'transaction.status',
 ];
 
-class GroupEvents extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    const queryParams = this.props.location.query;
-
-    this.state = {
-      query: queryParams.query || '',
-    };
-  }
-
-  componentDidMount() {
-    this._unsubscribeHandleRouteLeave = browserHistory.listen(newLocation =>
-      handleRouteLeave({
-        fieldsToClean: ['cursor'],
-        newLocation,
-        oldPathname: this.props.location.pathname,
-      })
-    );
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (this.props.location.search !== prevProps.location.search) {
-      const queryParams = this.props.location.query;
-
-      this.setState({query: queryParams.query});
-    }
-  }
-
-  componentWillUnmount() {
-    this._unsubscribeHandleRouteLeave?.();
-  }
-
-  _unsubscribeHandleRouteLeave: undefined | ReturnType<typeof browserHistory.listen>;
-
-  handleSearch = (query: string) => {
-    const targetQueryParams = {...this.props.location.query};
-    targetQueryParams.query = query;
-    const {organization} = this.props;
-    const {groupId} = this.props.params;
-
-    browserHistory.push(
-      normalizeUrl({
-        pathname: `/organizations/${organization.slug}/issues/${groupId}/events/`,
-        query: targetQueryParams,
-      })
-    );
-  };
-
-  render() {
-    return (
-      <Layout.Body>
-        <Layout.Main fullWidth>
-          <AllEventsFilters>
-            <EventSearchBar
-              organization={this.props.organization}
-              defaultQuery=""
-              onSearch={this.handleSearch}
-              excludedTags={excludedTags}
-              query={this.state.query}
-              hasRecentSearches={false}
-            />
-          </AllEventsFilters>
-          <AllEventsTable
-            issueId={this.props.group.id}
-            location={this.props.location}
-            organization={this.props.organization}
-            group={this.props.group}
+function GroupEvents({params, location, organization, group}: Props) {
+  const {groupId} = params;
+
+  useEffect(
+    () =>
+      browserHistory.listen(newLocation =>
+        handleRouteLeave({
+          fieldsToClean: ['cursor'],
+          newLocation,
+          oldPathname: location.pathname,
+        })
+      ),
+    [location]
+  );
```

Comment:
can we use `useCleanQueryParamsOnRouteLeave` here

Reply:
Is that a "please use this" or is it a "I dunno if it will work but maybe"

Reply:
i haven't used it yet but i know it exists. also use it

Reply:
@ryan953 did I use this rgiht

Reply:
Go add more comments :-)